lib/server.js

```diff
@@ -41,27 +41,39 @@ var Server = function(server, path, services, wsdl, options) {
   if (path[path.length - 1] !== '/')
     path += '/';
   wsdl.onReady(function(err) {
-    var listeners = server.listeners('request').slice();
-
-    server.removeAllListeners('request');
-    server.addListener('request', function(req, res) {
-      if (typeof self.authorizeConnection === 'function') {
-        if (!self.authorizeConnection(req)) {
-          res.end();
-          return;
+    if(typeof server.route === 'function' && typeof server.use === 'function'){
+      //handle only the required URL path for express server
+      server.route(path).all(function(req, res, next){
+        if (typeof self.authorizeConnection === 'function') {
+          if (!self.authorizeConnection(req)) {
+            res.end();
+            return;
+          }
         }
-      }
-      var reqPath = url.parse(req.url).pathname;
-      if (reqPath[reqPath.length - 1] !== '/')
-        reqPath += '/';
-      if (path === reqPath) {
         self._requestListener(req, res);
-      } else {
-        for (var i = 0, len = listeners.length; i < len; i++) {
-          listeners[i].call(this, req, res);
+      });
+    } else {
+      var listeners = server.listeners('request').slice();
+      server.removeAllListeners('request');
+      server.addListener('request', function(req, res) {
+        if (typeof self.authorizeConnection === 'function') {
+          if (!self.authorizeConnection(req)) {
+            res.end();
+            return;
+          }
         }
-      }
-    });
+        var reqPath = url.parse(req.url).pathname;
+        if (reqPath[reqPath.length - 1] !== '/')
```

Comment:
this snippet was taken from previous code, so, I wanted to leave it as it was to keep reviews simpler. But, will correct and squash the commits.